File: go/2101. Detonate the Maximum Bombs.go
func maximumDetonation(bombs [][]int) int {
    n := len(bombs)
    graph := make([][]int, n)
    for i := range graph {
        graph[i] = []int{}
    }
    for i := 0; i < n; i++ {
        for j := 0; j < n; j++ {
            if i != j && inRange(bombs[i], bombs[j]) {
                graph[i] = append(graph[i], j)
            }
        }
    }
    answer := 1
    for i := 0; i < n; i++ {
        answer = max(answer, dfs(graph, i, make([]bool, n)))
    }
    return answer
}

func dfs(graph [][]int, current int, visited []bool) int {
    count := 1
    visited[current] = true
    for _, i := range graph[current] {
        if visited[i] {
            continue
        }
        count += dfs(graph, i, visited)
    }
    return count
}

func inRange(a, b []int) bool {
    // returns true if b is in the range of a
    x1, y1, x2, y2 := a[0], a[1], b[0], b[1]
    return (x1-x2)*(x1-x2) + (y1-y2)*(y1-y2) <= a[2]*a[2]
}

File: java/2101. Detonate the Maximum Bombs.java
class Solution {
    public int maximumDetonation(int[][] bombs) {
        List<List<Integer>> graph = new ArrayList<>();
        for (int[] i : bombs) graph.add(new ArrayList<>());
        for (int i = 0; i < bombs.length; i++) {
            for (int j = 0; j < bombs.length; j++) {
                if (i != j && inRange(bombs[i], bombs[j])) {
                    graph.get(i).add(j);
                }
            }
        }
        int answer = 0;
        for (int i = 0; i < graph.size(); i++) {
            answer = Math.max(answer, dfs(graph, i, new boolean[graph.size()]));
        }
        return answer;
    }
    private int dfs(List<List<Integer>> graph, int current, boolean[] visited) {
        if (visited[current]) return 0;
        visited[current] = true;
        int answer = 1;
        for (int i : graph.get(current)) {
            if (visited[i]) continue;
            answer += dfs(graph, i, visited);
        }
        return answer;
    }
    private boolean inRange(int[] a, int[] b) {
        long x1 = a[0], x2 = b[0], y1 = a[1], y2 = b[1], r = a[2];
        long distance = (x1-x2)*(x1-x2) + (y1-y2)*(y1-y2);
        long range = r*r;
        return distance <= range;
    }
}